internal/kafka/internal/config/kafka_supported_instance_types.go

```diff
@@ -0,0 +1,202 @@
+		return fmt.Errorf("ingressThroughputPerSec for Kafka instance type '%s', size '%s' is invalid: %s", k.Id, instanceTypeId, err.Error())
+	}
+
+	maxDataRetentionSize, err := k.MaxDataRetentionSize.ToK8Quantity()
+	if err != nil {
+		return fmt.Errorf("maxDataRetentionSize for Kafka instance type '%s', size '%s' is invalid: %s", k.Id, instanceTypeId, err.Error())
+	}
+
+	maxDataRetentionPeriod, err := duration.ParseISO8601(k.MaxDataRetentionPeriod)
+	if err != nil {
+		return fmt.Errorf("maxDataRetentionPeriod for Kafka instance type '%s', size '%s' is invalid: %s", k.Id, instanceTypeId, err.Error())
+	}
+
+	if maxDataRetentionPeriod.IsZero() || egressThroughputQuantity.CmpInt64(1) < 0 ||
+		ingressThroughputQuantity.CmpInt64(1) < 0 || maxDataRetentionSize.CmpInt64(1) < 0 ||
+		k.TotalMaxConnections <= 0 || k.MaxPartitions <= 0 || k.MaxConnectionAttemptsPerSec <= 0 ||
+		k.QuotaConsumed < 1 || k.CapacityConsumed < 1 {
+		return fmt.Errorf("Kafka instance size '%s' for instance type '%s' specifies a property value less than or equals to Zero.", k.Id, instanceTypeId)
+	}
+
+	return nil
+}
+
+type SupportedKafkaInstanceTypesConfig struct {
+	SupportedKafkaInstanceTypes []KafkaInstanceType `yaml:"supported_instance_types"`
+}
+
+func (s *SupportedKafkaInstanceTypesConfig) validate() error {
+	existingInstanceTypes := make(map[string]int, len(s.SupportedKafkaInstanceTypes))
+
+	for _, KafkaInstanceType := range s.SupportedKafkaInstanceTypes {
+		if _, ok := existingInstanceTypes[KafkaInstanceType.Id]; ok {
+			return fmt.Errorf("Kafka instance type id '%s' was defined more than once.", KafkaInstanceType.Id)
+		}
+		existingInstanceTypes[KafkaInstanceType.Id]++
+
+		if err := KafkaInstanceType.validate(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+type KafkaSupportedInstanceTypesConfig struct {
+	Configuration     SupportedKafkaInstanceTypesConfig
+	ConfigurationFile string
+}
+
+func NewKafkaSupportedInstanceTypesConfig() *KafkaSupportedInstanceTypesConfig {
+	return &KafkaSupportedInstanceTypesConfig{
+		ConfigurationFile: "config/kafka-instance-types-configuration.yaml",
+	}
+}
+
+func (s *SupportedKafkaInstanceTypesConfig) GetKafkaInstanceTypeByID(instanceType string) (*KafkaInstanceType, error) {
+	for _, t := range s.SupportedKafkaInstanceTypes {
+		if t.Id == instanceType {
+			ret := t
+			return &ret, nil
+		}
+	}
+	return nil, fmt.Errorf("Unable to find kafka instance type for '%s'", instanceType)
+}
+
+func (kp *KafkaInstanceType) GetKafkaInstanceSizeByID(sizeId string) (*KafkaInstanceSize, error) {
+	for _, size := range kp.Sizes {
+		if size.Id == sizeId {
+			ret := size
+			return &ret, nil
+		}
+	}
+	return nil, fmt.Errorf("Kafka instance size id: '%s' not found for '%s' instance type", sizeId, kp.Id)
+}
+
+type Plan string
+
+func (p Plan) String() string {
+	return string(p)
+}
+
+func (p Plan) GetInstanceType() (string, error) {
+	t := strings.Split(strings.TrimSpace(p.String()), ".")
+	if len(t) != 2 {
```

Comment:
if this size of 2 is number of type of plans, better to have it as a constant.

Reply:
this refers to the format of the plan, it has to be `<plan-name>.<size-id>`, so two components. Not sure if this really needs to be a constant.

Reply:
It is not the number of plans but the number of components that there are in the format of a plan.

I don't see it strictly necessary but it does not harm anyway so I added a new commit with a constant for it.

Reply:
thanks Miguel!